task/opm-get-bundle-version/0.1/opm-get-bundle-version.yaml

```diff
@@ -0,0 +1,38 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: opm-get-bundle-version
+spec:
+  description: Fetch the current version of the provided OLM bundle image
+  params:
+    - name: bundle-image
+      description: OLM bundle image to query
+  results:
+    - name: bundle-version
+      description: olm.package version
+  workspaces:
+    - name: workspace
+      description: The shared workspace between steps
+  steps:
+    - name: opm-render-bundle
+      image: "registry.redhat.io/openshift4/ose-operator-registry:latest"
+      securityContext:
+        runAsUser: 0
```

Comment:
Why root is needed?

Reply:
hey @MartinBasti , I seem to get permission denied otherwise when trying to write to the workspace. Adam explained to me here that it is due to the git-clone task (https://github.com/konflux-ci/build-definitions/pull/1210#issuecomment-2257759876). Perhaps though there is another way to resolve this issue :smile_cat:  

```
❯ opc pipeline start --use-param-defaults --showlog --workspace name=workspace,volumeClaimTemplateFile=tkn/workspace-template.yaml catalog-dynamic-build
PipelineRun started: catalog-dynamic-build-run-xljnz
Waiting for logs to be available...
[init : init] Build Initialize: quay.io/redhat-user-workloads/oeo-cicada-tenant/osd-example-operator/catalog:on-pr-test
[init : init]
[init : init] Determine if Image Already Exists

[clone-repository : clone] {"level":"info","ts":1723102717.92717,"caller":"git/git.go:176","msg":"Successfully cloned https://github.com/openshift/osd-example-operator.git @ 587fc22f55c819208f91d719a40527a8bf913142 (grafted, HEAD) in path /workspace/output/source"}
[clone-repository : clone] {"level":"info","ts":1723102717.9459348,"caller":"git/git.go:215","msg":"Successfully initialized and updated submodules in path /workspace/output/source"}

[clone-repository : symlink-check] Running symlink check

[opm-get-bundle-version : opm-render-bundle] 2024/08/08 07:38:58 warning: unsuccessful cred copy: ".docker" from "/tekton/creds" to "/": unable to create destination directory: mkdir /.docker: permission denied
[opm-get-bundle-version : opm-render-bundle] + opm render quay.io/redhat-user-workloads/oeo-cicada-tenant/osd-example-operator/bundle@sha256:5485652c17ab8ac5cf44452f67e15ed06dc9d2d7d67a7f079e76248878d5370e
[opm-get-bundle-version : opm-render-bundle] /tekton/scripts/script-0-c69dr: line 3: /workspace/workspace/bundle.json: Permission denied

failed to get logs for task opm-get-bundle-version : container step-opm-render-bundle has failed  : [{"key":"StartedAt","value":"2024-08-08T07:39:20.186Z","type":3}]
```

Reply:
ah, okay